static/src/javascripts/projects/common/modules/identity/email-sign-in-banner/index.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+
+import { Message, hasUserAcknowledgedBanner } from 'common/modules/ui/message';
+import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import ophan from 'ophan/ng';
+import config from 'lib/config';
+import userPrefs from 'common/modules/user-prefs';
+import type { Banner } from 'common/modules/ui/bannerPicker';
+import { campaigns } from './campaigns';
+import { make as makeTemplate, messageCode } from './template';
+import type { Campaign } from './campaigns';
+
+const displayEventKey: string = `${messageCode} : display`;
+const userPrefsReferrerKey = 'emailbanner.referrerEmail';
+
+const emailPrefsLink = `https://${config.get('page.host')}/email-newsletters`;
+
+const signInLink = `${config.get('page.idUrl')}/login?returnUrl=${config.get(
+    emailPrefsLink
+)}`;
+
+const getEmailCampaignFromUtm = (utm): ?Campaign =>
```

Comment:
I get the idea of keeping campaigns.js as a pure data file but these utility functions being moved there would tidy up this file a bit (which is the one more likely to be read/updated in the future)

Reply:
nono you make a very solid case. Plus I normally like to only have `canShow`, `show`, and their immediate checks on the banner itself